core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/workflow/WorkflowResource.scala

```diff
@@ -414,22 +428,68 @@ class WorkflowResource extends LazyLogging {
     resultWorkflows.toList
   }
 
+  @GET
+  @Path("/content/{wid}")
+  def getWorkflowContentByWid(@PathParam("wid") wid: UInteger): String = {
+    val workflow = context
+      .selectFrom(WORKFLOW)
+      .where(WORKFLOW.WID.eq(wid))
+      .fetchOneInto(classOf[Workflow])
+
+    if (workflow == null) {
+      throw new WebApplicationException(s"Workflow with wid $wid not found.", 404)
+    }
+
+    val objectMapper = new ObjectMapper()
+    val contentMap =
+      objectMapper.readValue(workflow.getContent, classOf[java.util.Map[String, Any]])
+
+    val operators =
+      contentMap.get("operators").asInstanceOf[java.util.List[java.util.Map[String, Any]]]
+    val operatorInfoList = operators.asScala.map { operator =>
+      Map(
+        "operatorID" -> operator.get("operatorID"),
+        "operatorType" -> operator.get("operatorType")
+      ).asJava
+    }.asJava
+
+    objectMapper.writeValueAsString(operatorInfoList)
```

Comment:
what does this process do? it is not clear to me. it seems to be some kind of conversion.

Reply:
Since the frontend needs to retrieve the operator IDs that need to be replaced, this method returns all operator IDs of the cloned workflow. I felt that returning the entire workflow content would be too resource-intensive, so I opted to return only the necessary operator IDs and operator types.

Reply:
I suggest you return the workflow, as your method name is `getWorkflowContentByWid`, which means getting all content of a workflow.

Reply:
In fact, as we let frontend assign the new UUID logic, you don't need to change anything on the backend.

Reply:
Or should I change its name? I think we’ll only use this API to retrieve the operator IDs.